table/types/value_test.go

```diff
@@ -388,3 +392,424 @@ func TestNullable(t *testing.T) {
+					case src.len == dst.len && src.signed != dst.signed,
+						src.len > dst.len,
+						src.signed && !dst.signed:
+						mustErr = true
+					}
+					err := CastTo(OptionalValue(src.value), dst.destination)
+					if mustErr {
+						require.Error(t, err)
+					} else {
+						require.NoError(t, err)
+					}
+				})
+			}
+		})
+	}
+}
+
+func TestCastOtherTypes(t *testing.T) {
+	for _, tt := range []struct {
+		v      Value
+		dst    interface{}
+		result interface{}
+		error  bool
+	}{
+		{
+			v:      BytesValue([]byte("test")),
+			dst:    func(v []byte) *[]byte { return &v }(make([]byte, 0, 10)),
+			result: func(v []byte) *[]byte { return &v }([]byte("test")),
+			error:  false,
+		},
+		{
+			v:      TextValue("test"),
+			dst:    func(v []byte) *[]byte { return &v }(make([]byte, 0, 10)),
+			result: func(v []byte) *[]byte { return &v }([]byte("test")),
+			error:  false,
+		},
+		{
+			v:      BytesValue([]byte("test")),
+			dst:    func(v string) *string { return &v }(""),
+			result: func(v string) *string { return &v }("test"),
+			error:  false,
+		},
+		{
+			v:      DoubleValue(123),
+			dst:    func(v float64) *float64 { return &v }(9),
+			result: func(v float64) *float64 { return &v }(123),
+			error:  false,
+		},
+		{
+			v:      DoubleValue(123),
+			dst:    func(v float32) *float32 { return &v }(9),
+			result: func(v float32) *float32 { return &v }(9),
+			error:  true,
+		},
+		{
+			v:      FloatValue(123),
+			dst:    func(v float64) *float64 { return &v }(9),
+			result: func(v float64) *float64 { return &v }(123),
+			error:  false,
+		},
+		{
+			v:      FloatValue(123),
+			dst:    func(v float32) *float32 { return &v }(9),
+			result: func(v float32) *float32 { return &v }(123),
+			error:  false,
+		},
+		{
+			v:      Uint64Value(123),
+			dst:    func(v float32) *float32 { return &v }(9),
+			result: func(v float32) *float32 { return &v }(9),
+			error:  true,
+		},
+		{
+			v:      Uint64Value(123),
+			dst:    func(v float64) *float64 { return &v }(9),
+			result: func(v float64) *float64 { return &v }(9),
+			error:  true,
+		},
+		{
+			v:      OptionalValue(DoubleValue(123)),
+			dst:    func(v float64) *float64 { return &v }(9),
+			result: func(v float64) *float64 { return &v }(123),
+			error:  false,
+		},
+	} {
+		t.Run(fmt.Sprintf("cast %s to %v", tt.v.Type().String(), reflect.ValueOf(tt.dst).Type().Elem()),
+			func(t *testing.T) {
+				if err := CastTo(tt.v, tt.dst); (err != nil) != tt.error {
+					t.Errorf("castTo() error = %v, want %v", err, tt.error)
+				} else if !reflect.DeepEqual(tt.dst, tt.result) {
+					t.Errorf("castTo() result = %+v, want %+v",
+						reflect.ValueOf(tt.dst).Elem(),
+						reflect.ValueOf(tt.result).Elem(),
+					)
+				}
+			},
+		)
+	}
+}
+
+func TestValueToString(t *testing.T) {
+	for _, tt := range []struct {
+		value Value
+		exp   string
+	}{
+		{
+			value: VoidValue(),
+			exp:   "Void",
+		},
+		{
+			value: TextValue("foo"),
+			exp:   "Text(\"foo\")",
```

Comment:
Support https://pkg.go.dev/fmt#Formatter for this values.
Imho. String should be value only